Updates to improve input demo documentation (#4035)
PecanProject/pecan: models/peprmt/demo_run/01_run_pecan_peprmt.qmd

---
title: "Running PEPRMT Using PEcAn"
author: 
  - "Aritra Dey"

>   - "David LeBauer"

Reviewer: At this point you've modified the source demo enough that it's probably appropriate to list yourself as the sole author and credit  previous work with wording something like "This demo is patterned after PEcAn demo 1 (link here), which was written by Aritra Dey and David LeBauer". @AritraDey-Dev and @dlebauer does that make sense to y'all?

## Prerequisites

Before running this notebook, make sure you have:

- All the PEcAn packages installed. You can install all PEcAn packages and their dependencies by running the following command in the root of your PEcAn repository. This chunk does not run automatically since it will overwrite a version of PEcAn you installed by other methods. Run it by hand when needed.

```{r packages, eval = FALSE}

# Enable repository from pecanproject
options(repos = c(
  pecanproject = 'https://pecanproject.r-universe.dev',
  CRAN = 'https://cloud.r-project.org'))
# Download and install PEcAn.all in R
install.packages(c('PEcAn.all', 'PEcAn.PEPRMT'))

# Alternatively, install from GitHub
remotes::install_github("pecanproject/pecan", subdir = "base/all", 
                        ref = "develop", force = T)
remotes::install_github("pecanproject/pecan", subdir = "models/peprmt", 
                        ref = "develop")
```

> - A valid `pecan.xml` configuration file or use the example provided: `pecan/models/peprmt/demo_run/settings.xml`

Reviewer: I keep forgetting this is part of the "Before running this notebook...' list and reading it as a fragment. Would this rephrase help?
```suggestion
- You will also need a valid `pecan.xml` configuration file, such as the example provided: `pecan/models/peprmt/demo_run/settings.xml`
```

The directory structure created by PEcAn for this demo run will look like this:

```         
demo_outdir/               # Root output directory

> ├── run/                   # Configuration & execution metadata

Reviewer: ```suggestion
├── ensemble.samples.<ensemble_id>.Rdata # contains the selected parameter values for each ensemble member
├── run/                   # Configuration & execution metadata
```

│   ├── runs.txt           # List of run IDs (one per model realization)
│   ├── <runid>/           # Model-specific configs, including run job and driver data

> │   └── config.*           # Generated model configs (not used for PEPRMT)

Author: Is this true?

Reviewer: It's definitely true PEPRMT doesn't use it and TBH I'm not sure what files this is talking about or what model uses them. Probably safe to delete this line.

Reviewer: dropped in a1e2f79d1

├── out/                   # Raw model outputs by run ID

> │   └── <runid>/           # E.g., PEPRMT output files

Reviewer: ```suggestion
│   └── <runid>/           # E.g., PEPRMT output files
├── STATUS                 # timestamp and ok/error status of each workflow step
├── pecan.*.xml            # copies of the run settings saved at different steps
├── run_manifest.csv       # info about site, PFT, etc for each runid
└── samples.Rdata          # draws from parameter distributions

```

Reviewer: Rephrase all of the above freely, but key point is it seems worth documenting all these files here while we're explaining it.

│   └── config.*           # Generated model configs (not used for PEPRMT)
├── out/                   # Raw model outputs by run ID
│   └── <runid>/           # E.g., daily PEPRMT output files
│   └── <runid>/           # E.g., PEPRMT output files
```

The root output directory is defined here as `demo_outdir/` by `settings$outdir`. This directory contains log and record files from the PEcAn workflow. They provide a detailed record of how data was generated and are key components of the analysis metadata and provenance. These can be useful for debugging as well as for downstream analysis.

Key subdirectories include `run/` and `out/` that contain files used to configure and run the model, files generated by the underlying ecosystem model, and PEcAn standard outputs used in downstream analyses. These are described in subsequent sections.

> Additional outputs include logs, a `STATUS` file that records the steps of the workflow along with timestamps and whether each step was successful, and a copy of the `pecan.*.xml` file.

Author: We might want to add something like this back but as is, I found it more confusing than helpful

Reviewer: I think these can be added to the file tree diagram -- I'll leave some suggestions to illustrate
